### PR TITLE
Align the Configuration reference documentation with the code

### LINKED ISSUE #2913: [Improvement] Align the Configuration reference documentation with the code
The [Configuration reference document](https://gitversion.net/docs/reference/configuration) is out-of-date with respect to the code.

There are new configuration items that are not documented, such as `update-build-number`, and some of the configuration defaults documented for branches is not aligned with the code.

### PR DESCRIPTION
**Pull request by chris-codeflow**
## Description
- Aligned the configuration defaults with the code
- Added missing fields
- Fixed typos

## Related Issue
Closes #2913

## Motivation and Context
My motivation for submitting this PR was due to when I found that **GitVersion** was using a _pre-release_ label (tag) of `alpha` when calculating the SemVer in a `develop` branch, but the reference documentation stated the _pre-release_ label should be `unstable`.  After checking the latest source code in the `main` branch, I found that the documentation was not fully aligned with the source code.

## How Has This Been Tested?
Triple-checked the changes against the source code.

## Screenshots (if appropriate):
N/A

## Checklist:
- [x] My code follows the code style of this project.
- [] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/input/docs/reference/configuration.md [hunk 1]
```diff
@@ -28,21 +28,21 @@ GitHubFlow and GitFlow, probably with others too.
...
-You can run `GitVersion /showConfig` to see the effective configuration
-(defaults + overrides).
+To see the effective configuration (defaults and overrides), you can run
+`gitversion /showConfig`.
 
-To create your config file just type `GitVersion init` in your repo directory
-after installing via chocolatey and a minimal `GitVersion.yml` configuration
-file will be created. Modify this as you need.
+To create your config file just type `gitversion init` in your repo directory
+after installing via Chocolatey and a minimal `GitVersion.yml` configuration
```
Comment (asbjornu) [inline docs/input/docs/reference/configuration.md:35]: Make `installing` a reference link to `https://gitversion.net/docs/usage/cli/installation` so we don't have to mention Chocolatey explicitly here.

```suggestion
after [installing] and a minimal `GitVersion.yml` configuration
```

File: docs/input/docs/reference/configuration.md [hunk 2]
```diff
@@ -273,12 +271,16 @@ merge-message-formats:
 
 The regular expression should contain the following capture groups:
 
-*   SourceBranch - Identifies the source branch of the merge
-*   TargetBranch - Identifies the target of the merge
-*   PullRequestNumber - Captures the pull-request number
++ SourceBranch - Identifies the source branch of the merge
++ TargetBranch - Identifies the target of the merge
++ PullRequestNumber - Captures the pull-request number
```
Comment (asbjornu) [inline docs/input/docs/reference/configuration.md:276]: Why the change from `*` to `+` here?

```suggestion
+ `SourceBranch` - Identifies the source branch of the merge
+ `TargetBranch` - Identifies the target of the merge
+ `PullRequestNumber` - Captures the pull-request number
```

File: docs/input/docs/reference/configuration.md [hunk 3]
```diff
@@ -390,19 +412,19 @@ Take this commit graph
...
++ feature/foo branches off release/v1.0.0
+  + Branch release/v1.0.0 from main
+  + Branch feature/foo from release/v1.0.0
+  + Add a commit to both release/v1.0.0 and feature/foo
+  + release/v1.0.0 is the base for feature/foo
++ release/v1.0.0 branches off feature/foo
+  + Branch feature/foo from main
+  + Branch release/v1.0.0 from feature/foo
+  + Add a commit to both release/v1.0.0 and feature/foo
+  + feature/foo is the base for release/v1.0.0
```
Comment (asbjornu) [inline docs/input/docs/reference/configuration.md:424]: Why the change from `*` to `+`?

File: docs/input/docs/reference/configuration.md [hunk 4]
```diff
@@ -522,7 +544,7 @@ Indicates this branch config represents a release branch in GitFlow.
 ### is-mainline
 
 When using Mainline mode, this indicates that this branch is a mainline. By
-default support/ and main are mainlines.
+default *main* and *support* are mainlines.
```
Comment (asbjornu) [inline docs/input/docs/reference/configuration.md:547]:
```suggestion
default `main` and `support/*` are mainlines.
```

**Review by asbjornu — CHANGES_REQUESTED** [review]
Thanks for tidying up everything! I just have a few comments and suggestions.

File: docs/input/docs/reference/configuration.md (hunk 2 above)
Comment (chris-codeflow) [reply docs/input/docs/reference/configuration.md:276]: This wasn't an intended change; it was because I started my changes before commit  7809b809da2268e1d11ea55bf710d727674ef2d9.  I will revert and apply your suggestion.

File: docs/input/docs/reference/configuration.md (hunk 3 above)
Comment (chris-codeflow) [reply docs/input/docs/reference/configuration.md:424]: This wasn't an intended change; it was because I started my changes before commit 7809b809da2268e1d11ea55bf710d727674ef2d9. I will revert.

**Review by asbjornu — APPROVED** [review]

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by chris-codeflow** [comment]
When does the https://gitversion.net site get synced with the published documentation @asbjornu or @arturcic?

**Comment by arturcic** [comment]
It should be published on next planned release
